packages/central-server/src/apiV2/GETHandler/helpers.js

```diff
@@ -107,15 +139,24 @@ export const getQueryOptionsForColumns = (
       const resource = columnName.split('.')[0];
       const recordType = resourceToRecordType(resource);
       if (!recordTypesInQuery.has(recordType)) {
-        const joinCondition = customJoinConditions[recordType] || [
-          `${recordType}.id`,
-          `${baseRecordType}.${getForeignKeyColumnName(recordType)}`,
-        ];
-        multiJoin.push({ joinWith: recordType, joinCondition, joinType });
-        recordTypesInQuery.add(recordType);
+        const join = customJoinConditions[recordType];
+        if (join) {
+          addJoin(
+            recordType,
+            recordTypesInQuery,
+            multiJoin,
+            baseRecordType,
+            customJoinConditions,
+            joinType,
+          );
+        }
       }
     });
+  console.log(recordTypesInQuery.size);
   // Ensure every join table is added to the sort, so that queries are predictable during pagination
+  console.log([...recordTypesInQuery]);
   const sort = [...recordTypesInQuery].map(recordType => `${recordType}.id`);
+  console.log(sort);
+  console.log(multiJoin);
```

Comment:
```suggestion
  // Ensure every join table is added to the sort, so that queries are predictable during pagination
  const sort = [...recordTypesInQuery].map(recordType => `${recordType}.id`);
```